torch/optim/adam.py

```diff
@@ -316,13 +316,16 @@ def adam(params: List[Tensor],
     # and when differentiable=False.
     # We still respect when the user inputs False for fused.
     if fused is None:
-        if not differentiable and all(
-            p.is_cuda and torch.is_floating_point(p)
-            for p in params + grads + exp_avgs + exp_avg_sqs + max_exp_avg_sqs + state_steps
-        ):
-            fused = True
-        else:
-            fused = False
+        all_tensors = []
+        all_tensors.extend(params)
+        all_tensors.extend(grads)
+        all_tensors.extend(exp_avgs)
+        all_tensors.extend(exp_avg_sqs)
+        all_tensors.extend(max_exp_avg_sqs)
+        all_tensors.extend(state_steps)
```

Comment:
Couldn't you refactor into a nice single itertools.chain, that way you don't need any temporary list at all.

Reply:
I started with that, but it does not play well with JIT bindings: https://github.com/pytorch/pytorch/pull/91896#discussion_r1069858978

Reply:
I see, which is annoying because jit bindings shouldn't be necessary since this path isn't part possible under JIT.

Reply:
The seven lines can be changed into just 1 line, no?
`all_tensors = [*params, *grads, *exp_avgs, *exp_avg_sqs, *max_exp_avg_sqs, *state_steps]`